inductiva/api/methods.py

```diff
@@ -331,6 +332,10 @@ def invoke_async_api(method_name: str,
 
         if resource_pool_id is None:
             logging.info("Task submitted to the default resource pool.")
+            if inductiva.resources.machine_groups.get():
```

Comment:
I am not sure if this API call should be in the middle of a simulation workflow. 
This seems like babysitting the users. I think we should provide the tool more clearly, for example, in the CLI so that they are aware of their active machine groups.

This seems to be warning for the mistake of forgetting to add the `machine_group/on` parameter in the simulator. Then, we probably should just make the message above more "visible", so that they are reminded of that in their workflow.

Reply:
So you are suggesting, for example:

```python
    def run(
        self,
        input_dir: types.Path,
        *_args,
        machine_group: Optional[resources.MachineGroup] = None,
        storage_dir: Optional[types.Path] = "",
        extra_metadata: Optional[dict] = None,
        **kwargs,
    ) -> tasks.Task:

        input_dir = self._setup_input_dir(input_dir)

        # Warn user here

        return tasks.run_simulation(
            self.api_method_name,
            input_dir,
            machine_group=machine_group,
            storage_dir=storage_dir,
            extra_metadata=extra_metadata,
            **kwargs,
        )

```

Reply:
What I was trying to say is that we shouldn't make calls to check if there are machines active in the middle of the simulation workflow. Think like this, should we also warn the user if he is launching a new machine group, when he has others active?

I think the user is responsible for this, otherwise we would be responsible for warning of various other stuff.

Reply:
Some interesting points here. It is useful to inform the user that he has allocated resources that he is not using, but I understand that it might become to noisy to inform him about that every time.

I suggest extracting this logic to an external function that gets called everytime, but the call to `inductiva.resources.machine_groups.get()` only gets made once and the warning is only emitted the first time the user runs a task on the default pool even though he has resources allocated. The message should be clear about what is being checked and that the warning will be silenced from that point onwards

Reply:
So we would like something like this:

```python
simulator = SomeSimulator(...)
simulator.simulate(...)  # -> This warns the user
simulator.simulate(...). # -> This no longer warns the user

simulator = AnotherSimulator(...)
simulator.simulate(...)  # -> This warns the user AGAIN
simulator.simulate(...). # -> This no longer warns the user
```

Or would we like the user to be warned only once even though a new simulator was just created. 

```python
simulator = SomeSimulator(...)
simulator.simulate(...)  # -> This warns the user ONLY here
simulator.simulate(...)

simulator = AnotherSimulator(...)
simulator.simulate(...)
simulator.simulate(...)
```

In the first case we could simply go about setting an internal variable on the simulator. Something like: `self.checked_resource_pool`. In the second case we could make use of environment variables (which would also be exposed to the user).